Example/Emission/AppDelegate.m

```diff
@@ -98,27 +99,41 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
 
 #pragma mark - Emission
 
+- (void)reloadEmission
+{
+  AppSetup *setup = [AppSetup ambientSetup];
+  NSString *service = setup.inStaging ? @"Emission-Staging" : @"Emission-Production";
+
+  BOOL isImpersonating = UserID && UserAccessToken;
+  AuthenticationManager *auth = isImpersonating ? nil : [[AuthenticationManager alloc] initWithService:service];
+
+  NSString *userID = isImpersonating ? UserID : auth.userID;
+  NSString *accessToken = isImpersonating ? UserAccessToken : auth.token;
+  [self setupEmissionWithUserID:userID accessToken:accessToken keychainService:service];
+}
+
 - (void)setupEmissionWithUserID:(NSString *)userID accessToken:(NSString *)accessToken keychainService:(NSString *)service;
 {
   AREmission *emission = nil;
 
   AppSetup *setup = [AppSetup ambientSetup];
 
   EmissionKeys *keys = [[EmissionKeys alloc] init];
-
-  AREmissionConfiguration *config = [[AREmissionConfiguration alloc]
-                                     initWithUserID:userID
-                                     authenticationToken:accessToken
-                                     sentryDSN:nil
-                                     stripePublishableKey:[keys stripePublishableKey]
-                                     googleMapsAPIKey:nil
-                                     gravityURL:setup.gravityURL
-                                     metaphysicsURL:setup.metaphysicsURL
-                                     predictionURL:setup.predictionURL
-                                     userAgent:@"Emission Example"];
-
+  
+  AREmissionConfiguration *config = [[AREmissionConfiguration alloc] initWithUserID:userID
+                                                                authenticationToken:accessToken
+                                                                          sentryDSN:nil
+                                                               stripePublishableKey:[keys stripePublishableKey]
+                                                                   googleMapsAPIKey:nil
+                                                                         gravityURL:setup.gravityURL
+                                                                     metaphysicsURL:setup.metaphysicsURL
+                                                                      predictionURL:setup.predictionURL
+                                                                          userAgent:@"Emission Example"
+                                                                            options:setup.options];
+  
```

Comment:
The `:`'s being aligned is how obj-c is supposed to handle these long method calls. I believe the aim is to balance the label and values. I dunno.